java-connectors/kafka-connect-azure-servicebus/src/test/java/io/lenses/streamreactor/connect/azure/servicebus/source/TaskToReceiverBridgeTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.azure.servicebus.source;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.storage.OffsetStorageReader;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class TaskToReceiverBridgeTest {
+
+  Map<String, String> properties = new HashMap<>();
+  ArrayBlockingQueue<ServiceBusMessageHolder> blockingQueue;
+  OffsetStorageReader offsetReader;
+
+  TaskToReceiverBridge testObj;
+
+  @BeforeEach
+  void setUp() {
+    blockingQueue = mock(ArrayBlockingQueue.class);
+    offsetReader = mock(OffsetStorageReader.class);
```

Comment:
This causes an unsafe cast warning.

My suggestion is to use @ExtendWith(MockitoJUnitExtension.class) and then you can annotate the fields as `@Mock`.  Then you can probably remove the `setUp` method.

Reply:
But underneath it would've done the same thing, wouldn't it? It'd just have runner on top of it.

It's just how Mockito works 🙂

Reply:
This is discipline to avoid the errors/violations/problems appearing in the IDE/static analysis tools, so we can focus on the real problems without adding extra noise.

Reply:
Changed, but it's a bit like sweeping not-really-existent problem under the rug